### PR TITLE
NODE-1525: generate documentation, embed in README

### PR DESCRIPTION
**Pull request by mbroadst**

### CODE REVIEW

File: lib/kerberos.js [hunk 1]
```diff
@@ -4,10 +4,6 @@ const kerberos = require('bindings')('kerberos');
 const KerberosClient = kerberos.KerberosClient;
 const KerberosServer = kerberos.KerberosServer;
 
-// Result Codes
-const AUTH_GSS_CONTINUE = 0;
```
Comment (daprahamian) [inline lib/kerberos.js:8]: Why are these removed?

**Review by daprahamian — APPROVED** [review]
👍 only one question

File: README.md [hunk 2]
```diff
@@ -1,70 +1,227 @@
...
+It does this by checking that the supplied user name and password can be
+used to get a ticket for the supplied service.
+If the user name does not contain a realm, then the default realm supplied
+is used.</p>
+<p>For this to work properly the Kerberos must be configured properly on this
+machine.
+That will likely mean ensuring that the edu.mit.Kerberos preference file
+has the correct realms and KDCs listed.</p>
+<p>IMPORTANT: This method is vulnerable to KDC spoofing attacks and it should
+only used for testing. Do not use this in any production system - your
```
Comment (kvwalker) [inline README.md]: 'only used' => 'only be used'

File: README.md [hunk 3]
```diff
@@ -1,70 +1,227 @@
...
+If the user name does not contain a realm, then the default realm supplied
+is used.
+
+For this to work properly the Kerberos must be configured properly on this
+machine.
+That will likely mean ensuring that the edu.mit.Kerberos preference file
+has the correct realms and KDCs listed.
+
+IMPORTANT: This method is vulnerable to KDC spoofing attacks and it should
+only used for testing. Do not use this in any production system - your
```
Comment (kvwalker) [inline README.md]: 'be' before used

**Review by kvwalker — APPROVED** [review]
looks good other than 2 small comments

File: lib/kerberos.js (hunk 1 above)
Comment (mbroadst) [reply lib/kerberos.js:8]: these were throwbacks from the way the C-style api worked, each time you would call a method it would return one of these results, and then the user would use an accessor to grab the response. Instead, I chose to return the actual response or error on failure. It's analagous to:
```
client.step(..., (err, res) => {
  if (res === AUTH_GSS_ERROR) return client.getError();
  client.getResponse()
});
```
vs
```
client.step(..., (err, response) => {
  if (err) return err;
  // we already have the response ready
});
```
